src/t.jl

```diff
@@ -28,18 +28,18 @@ export OneSampleTTest, TwoSampleTTest, EqualVarianceTTest,
 @compat abstract type TTest <: HypothesisTest end
 @compat abstract type TwoSampleTTest <: TTest end
 
-pvalue(x::TTest; tail=:both) = pvalue(TDist(x.df), x.t; tail=tail)
-
-default_tail(test::TTest) = :both
+pvalue(x::TTest; tail=x.tail) = pvalue(TDist(x.df), x.t; tail=tail)
+tail(x::TTest)  = x.tail    # defaults set by constructors
```

Comment:
Hi @nalimilan,

Adding a `tail` field to all types, which is not used by several of them, is not a quite clean solution. I would rather add the field to those types that actually can make use of it. These are quite many, though: Below, you find a list of types for which the `pvalue` function has a `tail` parameter, and those types for which `pvalue` has a fixed tail.

I will add a `tail` field to those types, for which the `pvalue` function takes a tail parameter.

The alpha value is relevant to all tests. Therefore, I will add an `alpha` field to all types. I will also add a simple function obtaining the test outcome:
```julia
reject_h0(test::HypothesisTest) = pvalue(test) <= test.alpha
```

I am looking forward to your comments!

---

`pvalue` with fixed tail (no tail parameter; 9 types):
- OneSampleADTest
- KSampleADTest
- ADFTest
- BoxPierceTest
- LjungBoxTest
- BreuschGodfreyTest
- RayleighTest
- JarqueBeraTest
- KruskalWallisTest

`pvalue` with tail parameter (15 types):
- BinomialTest
- SignTest
- FisherTLinearAssociation
- JammalamadakaCircularCorrelation
- FisherExactTest
- ExactKSTest
- ApproximateOneSampleKSTest
- ApproximateTwoSampleKSTest
- ExactMannWhitneyUTest
- ApproximateMannWhitneyUTest
- PowerDivergenceTest
- ExactSignedRankTest
- ApproximateSignedRankTest
- ZTest
- TTest